common/utils.py

```diff
@@ -132,3 +134,33 @@ def ensure_minimal_schema_version():
                         return
                     LOGGER.warning("Current schema version: %s, minimal required: %s, waiting...", current_schema, MINIMAL_SCHEMA)
                     sleep(10)
+
+
+def patch_get_request(endpoint, session=None):
```

Comment:
I'd somehow re-factor the vmaas_post_request method and use it from all places. It has a lot of common code.